tgui/packages/tgui/interfaces/AiAirlock.js

```diff
@@ -0,0 +1,202 @@
+import { Fragment } from 'inferno';
+import { useBackend } from '../backend';
+import { Button, LabeledList, Section } from '../components';
+import { Window } from '../layouts';
+
+const dangerMap = {
+  2: {
+    color: 'good',
+    localStatusText: 'Offline',
+  },
+  1: {
+    color: 'average',
+    localStatusText: 'Caution',
+  },
+  0: {
+    color: 'bad',
+    localStatusText: 'Optimal',
+  },
+};
+
+export const AiAirlock = (props, context) => {
+  const { act, data } = useBackend(context);
```

Comment:
Rather than access every field from `data` by doing `data.power` etc., you can make local variables from them by doing: 
```js
const {
	power,
	wires,
	// ETC...
} = data;
```

Reply:
I could, but I don't really see any advantage to doing that, aside from slightly shorter var names, which isn't really worth the price of having every var name be different from TG.

Reply:
They wouldn't be different, just made local. `data.power` would turn into `power`. Besides, I feel like we shouldn't copy everything from TG as-is. Porting is fine, but it should also be an opportunity to improve the code that is being ported, no matter how insignificant the problem might be. It's a matter of good practice within the codebase and consistency.